### PR TITLE
Update ggez-goodies to 0.9.0

### PR DESCRIPTION
**Pull request by Vixeliz**
This is basically a complete update to 0.9.0 don't merge yet though. The input example isn't fully working.

### CODE REVIEW

File: examples/scene_test.rs [hunk 1]
```diff
@@ -1,182 +1,125 @@
...
-            name: name.to_string(),
-            next_scene: next_scene.to_string(),
+impl Scene<SharedState> for StartScene {
+    fn update(
+        &mut self,
+        _: &mut SharedState,
+        ctx: &mut ggez::Context,
+    ) -> SceneSwitch<SharedState, SceneEvents, ggez::Context> {
+        if ctx.keyboard.is_key_just_pressed(VirtualKeyCode::K) {
+            println!("Hello");
```
Comment (sampullman) [inline examples/scene_test.rs]: This should probably be removed.

Also, the previous version used "space" to switch scenes and printed that in the display, but it's not a big deal.

Comment (Vixeliz) [reply examples/scene_test.rs:51]: What do you mean? The println!() or the whole scene i'm a little confused

Comment (Vixeliz) [reply examples/scene_test.rs]: Do you mean the println!() or something else?

Comment (Vixeliz) [reply examples/scene_test.rs]: Cleaned up the SceneSwitch generics and removed the println

**Pull request merged**